lesson5/main.js

```diff
@@ -7,10 +7,8 @@ const listItems = [
   { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" },
 ];
 
-const newList = new Promise((resolve) => {
-  resolve(listItems);
-}).then((val) => {
-  val.forEach((key) => {
+function newList(date) {
+  date.forEach((key) => {
```

Comment:
The following three processing can be divided into functions or variables.

```
new Promise((resolve, reject) => {
  resolve(listItems);
})
```
↑ Please refer to this Link.
（The ideal is looking like the sample code. reject is OK. Its fine as is.）
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

```
data.forEach((key) => {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");

    a.href = `/${key.to}`;
    a.textContent = key.text;
    img.src = key.img;
    img.alt = key.alt;

    fragment
      .appendChild(li)
      .appendChild(a)
      .insertAdjacentElement("afterbegin", img);
});
```
↑ ①：This processing can be function as returning `fragment` .

```
const ul = document.getElementById("js-lists");
ul.appendChild(fragment);
```
↑ ②：set created by ① at args. 
And This also can also be created as a function.